### ISSUE TITLE
On Arch linux 'stack install' succeeds while 'stack ghci' doesn't

### ISSUE THREAD

**Issue by PierreR** [commercialhaskell/stack#1135]
I am trying to use `stack ghci` with language-puppet. `stack install` works fine but `stack ghci` gives me a compilation error. For instance:
```
Puppet/Utils.hs:30:40:
    No instance for (AsNumber Scientific)
      arising from a use of ‘_Integer’
```
This error makes no sense to me because
- the package does build using `stack build`
- the lens-aeson version used in the plan [does include such instance](http://hackage.haskell.org/package/lens-aeson-1.0.0.5/docs/Data-Aeson-Lens.html).

It is important to note that:
- I have been tested the same setting on `Ubuntu 14.04.02 LTS Trusty Tahr` and `stack ghci` works fine there.
- I have other (simpler) projects on Arch linux that have no problem with `stack ghci`.
- After a `stack clean`, I have another error: `Could not find module ‘Paths_language_puppet’`. When I `stack build` again and then `stack ghci`, I will get back the same compilation error.
- Trying to remove `~/.ghci` does not make a difference

One notable difference between the Ubuntu and the Arch linux setup, is the fact that in Ubuntu everything is installed by `stack` whereas in `Arch linux`, ghc and many other haskell packages are installed by Arch linux (via pacman)

How to reproduce (on Arch linux):
- git clone https://github.com/bartavelle/language-puppet.git
- stack install (all good)
- stack ghci (compilation error as described).

Using Version 0.1.5.0 X86_64 on  Arch linux.

As already discussed in #847, the problem is related to Arch linux. It is suggested in #847 that permissions  could be the root cause. I have tested the proposed fix (make file permissions on Arch the same as in Ubuntu) but it did not fix the problem.

**Comment by borsboom** [comment]
What happens if you use `stack --no-system-ghc`, so that Stack doesn't use the system-installed GHC/packages?  

Also, AFAIK, there's no inherent reason Stack on Arch has to use the system GHC/package.  It would only do so if you have already installed GHC/packages separately from Stack (just like on Ubuntu and others).  The haskell-stack package at https://aur.archlinux.org/packages/haskell-stack/ does not depend on any system GHC (but will use it if it's there).

**Comment by borsboom** [comment]
Also, @drwebb, as our resident Arch user, do you have input about why the system-installed GHC/packages would be causing this kind of trouble with `stack ghci`?

**Comment by PierreR** [comment]
As a note, I will have to check to see if  0a1a2f75bc29c02349335a68e488d8fecd73156e does not solve the issue. 

@borsboom If it does not, I will definitely try `stack --no-system-ghc`

**Comment by PierreR** [comment]
I can confirm this issue is solved in  `0.1.7.0`.

Thanks for the support.

**Comment by drwebb** [comment]
I've never come across a problem where using the ghc package in the Arch
repos acted like this.

Emanuel Borsboom writes:

> Also, @drwebb, as our resident Arch user, do you have input about why the system-installed GHC/packages would be causing this kind of trouble with `stack ghci`?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/commercialhaskell/stack/issues/1135#issuecomment-147408023

-- 
Tristan Webb PhD
Application Engineer
FP Complete
